pkg/signing/signing.go

```diff
@@ -274,6 +275,50 @@ func getSigsPrimPart(fimg *sif.FileImage) (sigs []*sif.Descriptor, descr []*sif.
 	return
 }
 
+func getSigsAllPart(fimg *sif.FileImage) ([]*sif.Descriptor, []*sif.Descriptor, error) {
+	var dpart []*sif.Descriptor
+	var sigs []*sif.Descriptor
+	var err error
+
+	// Ensure theres a primary partition.
+	_, _, err = fimg.GetPartPrimSys()
+	if err != nil {
+		return nil, nil, fmt.Errorf("no primary partition found")
+	}
+
+	// Loop through all the partitions, (skipping DataSignatures)
+	// and collect all the signatures for a data partition.
+	for _, d := range fimg.DescrArr {
+		if !d.Used {
+			continue
+		}
+		// No need to verify a signature.
+		if d.Datatype == sif.DataSignature {
+			continue
+		}
+		s, _, err := fimg.GetLinkedDescrsByType(d.ID, sif.DataSignature)
+		if err != nil {
+			// If a partition is not signed, print a warning.
+			sylog.Warningf("No signature found for %s\n", datatypeStr(d.Datatype))
```

Comment:
Are you sure we should print a warning here? E.g. I haven't signed some JSON config, IMO that is not something to consider (maybe I am wrong here). Also when we do `singularity sign` does it sign all partitions? If not, that will lead to lots of false warnings.
p.s. no need for new line in logs (it is added automatically)

Reply:
Yes, sign will now sign all partitions, as of https://github.com/sylabs/singularity/pull/4477. I think there should be a warning here. If not a warning, what do you think it should be?

Reply:
A warning for every non-signed partition generated with an earlier version might be annoying. What about at the end we say something more like...

`system partition verified.....`

`all partitions verified....`

?

Reply:
I'm okay with `All partitions verified` after the verify output, but I kinda like the warning here, it give the user a general idea on what partitions are there, and whats signed/unsigned. But I'll change it if you prefer something else.

Reply:
I don't mind a warning per descriptor. I kind of mind the format of the message. I would prefer something like:

```
WARNING: Missing signature for SIF descriptor 1 (Def.FILE)
```

It will create questions from users, yes (mostly in the form of "do I care?"). The danger with this sort of message is that you are training users to ignore them, but since by default there are going to be only a handful of these lines, I think it's OK.

I would output a INFO-level message like:

```
INFO: All SIF descriptors verified
```

just to let users know that everything is OK.

Over time the ecosystem will transition from the warnings to this last message (as images are updated).